osgp/protocol-adapter-dlms/osgp-protocol-simulator-dlms/simulator/dlms-device-simulator/src/main/java/org/opensmartgridplatform/simulator/protocol/dlms/cosem/PowerQualityThdEventLog.java

```diff
@@ -0,0 +1,153 @@
+      snOffset = 0x08,
+      selector = {1, 2})
+  private final DataObject buffer;
+
+  /**
+   * Only for cosem attribute definition, data remains untouched. Attribute data is gathered from
+   * {@link #CAPTURE_OBJECT_DEFINITIONS}
+   */
+  @CosemAttribute(
+      id = 3,
+      type = Type.ARRAY,
+      accessMode = AttributeAccessMode.READ_ONLY,
+      snOffset = 0x10)
+  private final DataObject captureObjects;
+
+  @CosemAttribute(
+      id = 4,
+      type = Type.DOUBLE_LONG_UNSIGNED,
+      accessMode = AttributeAccessMode.READ_ONLY,
+      snOffset = 0x18)
+  public DataObject capturePeriod;
+
+  @CosemAttribute(
+      id = 5,
+      type = Type.ENUMERATE,
+      accessMode = AttributeAccessMode.READ_ONLY,
+      snOffset = 0x20)
+  public DataObject sortMethod;
+
+  @CosemAttribute(
+      id = 6,
+      type = Type.STRUCTURE,
+      accessMode = AttributeAccessMode.READ_ONLY,
+      snOffset = 0x28)
+  public DataObject sortObject;
+
+  /**
+   * Only for cosem attribute definition, data remains untouched. Attribute data is gathered from
+   * size of {@link #bufferData}
+   */
+  @CosemAttribute(
+      id = 7,
+      type = Type.DOUBLE_LONG_UNSIGNED,
+      accessMode = AttributeAccessMode.READ_ONLY,
+      snOffset = 0x30)
+  private final DataObject entriesInUse;
+
+  @CosemAttribute(
+      id = 8,
+      type = Type.DOUBLE_LONG_UNSIGNED,
+      accessMode = AttributeAccessMode.READ_ONLY,
+      snOffset = 0x38)
+  public DataObject profileEntries;
+
+  public PowerQualityThdEventLog(final Calendar time) {
+    super("0.0.99.98.8.255");
+    this.time = time;
+
+    this.buffer = DataObject.newNullData();
+    this.captureObjects = DataObject.newNullData();
+    this.capturePeriod = DataObject.newUInteger32Data(CAPTURE_PERIOD);
+    this.sortMethod = DataObject.newEnumerateData(SortMethod.FIFO.value());
+    this.sortObject = DataObject.newNullData();
+    this.entriesInUse = DataObject.newNullData();
+    this.profileEntries = DataObject.newUInteger32Data(PROFILE_ENTRIES);
+
+    this.initBufferData();
+  }
+
+  private static CaptureObjectDefinitionCollection initCaptureObjects() {
+    final CaptureObjectDefinitionCollection definitions = new CaptureObjectDefinitionCollection();
+
+    // Timestamp
+    definitions.add(
+        new CaptureObjectDefinition(
+            new CaptureObject(8, "0.0.1.0.0.255", (byte) 2, 0), new CosemDateTimeProcessor()));
+
+    // Eventcode
+    definitions.add(
+        new CaptureObjectDefinition(
+            new CaptureObject(1, "0.0.96.11.8.255", (byte) 2, 0), new UInteger16DataProcessor()));
+
+    // Magnitude
+    definitions.add(
+        new CaptureObjectDefinition(
+            new CaptureObject(1, "0.0.96.11.22.255", (byte) 2, 0), new UInteger16DataProcessor()));
+
+    return definitions;
+  }
+
+  /** Initializes buffer with some data. */
+  private void initBufferData() {
+    this.bufferData = new CircularFifoQueue<>(PROFILE_ENTRIES);
+
+    // Add all events.
```

Comment:
comments really necessary?

Reply:
Removed commentline. Made method documentation more descriptive. Renamed parameters of method addEvents more descriptive.